src/test/kotlin/com/exactpro/th2/check1/CollectorServiceTest.kt

```diff
@@ -0,0 +1,117 @@
+package com.exactpro.th2.check1
+
+import com.exactpro.th2.check1.configuration.Check1Configuration
+import com.exactpro.th2.check1.grpc.CheckRuleRequest
+import com.exactpro.th2.common.event.EventUtils
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.MessageBatch
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.ConnectionID
+import com.exactpro.th2.common.grpc.RootMessageFilter
+import com.exactpro.th2.common.grpc.Direction
+import com.exactpro.th2.common.grpc.Checkpoint
+import com.exactpro.th2.common.message.toTimestamp
+import com.exactpro.th2.common.schema.message.MessageRouter
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import kotlin.system.measureTimeMillis
+
+class CollectorServiceTest {
+    private val eventRouterStub: MessageRouter<EventBatch> = mock {}
+    private val messageRouterStub: MessageRouter<MessageBatch> = mock<MessageRouter<MessageBatch>> {}
+        .apply { whenever(subscribeAll(any())).thenReturn(mock()) }
```

Comment:
I think it is better to do this way
```suggestion
    private val messageRouterStub: MessageRouter<MessageBatch> = mock<MessageRouter<MessageBatch>> {
      on { subscribeAll(any()) }.thenReturn(mock())
    }
```